src/ServiceInsight/Framework/Settings/ApplicationConfiguration.cs

```diff
@@ -0,0 +1,27 @@
+﻿namespace ServiceInsight.Framework.Settings
+{
+    using System;
+    using System.Configuration;
+
+    public class ApplicationConfiguration
+    {
+        public static bool SkipCertificateValidation { get; private set; }
+
+        public static void Initialize()
+        {
+            SkipCertificateValidation = GetSkipCertificateValidation();
+        }
+
+        private static bool GetSkipCertificateValidation()
+        {
+            try
+            {
+                return Convert.ToBoolean(ConfigurationManager.AppSettings["SkipCertificateValidation"]);
+            }
+            catch (Exception ex)
+            {
+                throw new ApplicationException("Application setting 'SkipCertificateValidation' is either missing or cannot be converted to type boolean.", ex);
```

Comment:
`ApplicationException` is effectively deprecated. See https://docs.microsoft.com/en-gb/dotnet/api/system.applicationexception?view=netframework-4.7 (emphasis mine):

> ⚠️ Important
> You should derive custom exceptions from the Exception class rather than the ApplicationException class. **You should not throw an ApplicationException exception in your code**, and you should not catch an ApplicationException exception unless you intend to re-throw the original exception.

I think the correct type to use could be [`ConfigurationErrorsException`](https://docs.microsoft.com/en-gb/dotnet/api/system.configuration.configurationerrorsexception?view=netframework-4.7) (using the same `string`, `Exception` constructor).